### PR TITLE
Add 203 profile

### PR DESCRIPTION
**Pull request by abrooksv**
## Description

* Add 203 profiles
* Add run configs
* Fix a compile error in 203

There are still 2 compile issues in jetbrains-core, other modules unknown
Also rdgen for 203 changes Gradle API so we can't use it. (Newer versions seems to compile in a way that keeps 201 and 202 working (but not 193) though if we use 203 for all of them). Needs more investigation

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by abrooksv** [comment]
The renaming issues do not look to be easy to fix and will probably need to add shims in for 201 and 202, 203+

e: aws-toolkit-jetbrains/jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/java/JavaSamDebugSupport.kt: (33, 40): Incompatible types: XDebugSessionImpl and DebuggerSession
e: aws-toolkit-jetbrains/jetbrains-core/src/software/aws/toolkits/jetbrains/services/sqs/actions/PurgeQueueAction.kt: (55, 30): Using 'show(): Int' is an error. Use ask(project)

**Comment by abrooksv** [comment]
Resharper backend compiles against 203,

but Kotlin front end generated code is not aligning anymore

203:
`import com.jetbrains.rd.ide.model.awsProjectModel`

201,202:
`import com.jetbrains.rider.model.awsProjectModel`

**Comment by abrooksv** [comment]
Rebased on master branch,

TODO:
* Add wrapper for TestDialog migration for tests and create source set shims for them
* Add source sets for 201,202 for changes in the dotnet wizard

**Comment by abrooksv** [comment]
Core tests pass now after #2190 

TODO:
    Add source sets for 201,202 for changes in the dotnet wizard

**Comment by abrooksv** [comment]
Moved our RD protocol models into our namespace and out of JetBrains', Rider unit tests fail though:

```
  java.lang.ClassCastException: class software.aws.toolkits.jetbrains.settings.MockAwsSettings cannot be cast to class software.aws.toolkits.jetbrains.settings.AwsSettings (software.aws.toolkits.jetbrains.settings.MockAwsSettings is in unnamed module of loader 'app'; software.aws.toolkits.jetbrains.settings.AwsSettings is in unnamed module of loader com.intellij.ide.plugins.cl.PluginClassLoader @31cb3ad9)
      at software.aws.toolkits.jetbrains.settings.AwsSettings$Companion.getInstance(AwsSettings.kt:28)
      at software.aws.toolkits.jetbrains.core.AwsTelemetryPrompter.runActivity(AwsTelemetryPrompter.kt:26)
      at com.intellij.ide.startup.impl.StartupManagerImpl.runActivity(StartupManagerImpl.java:301)
```

TODO:
* Add source sets for 201,202 for changes in the dotnet wizard
* Get Rider tests to run

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/DownloadLogStream.kt [hunk 1]
```diff
@@ -140,7 +141,7 @@ class LogStreamDownloadToFileTask(
         val descriptor = FileSaverDescriptor(message("cloudwatch.logs.download"), message("cloudwatch.logs.download.description"))
         val saveLocation = withContext(edt) {
             val destination = FileChooserFactory.getInstance().createSaveFileDialog(descriptor, project)
-            destination.save(null, logStream)
+            destination.save(null as VirtualFile?, logStream)
```
Comment (hunterwerlla) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/DownloadLogStream.kt:144]: Does this need a fix when min?

File: jetbrains-core/tst-201-202/software/aws/toolkits/jetbrains/ui/TestDialogService.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.ui
+
+import com.intellij.openapi.ui.Messages
+import com.intellij.openapi.ui.TestDialog
+import com.intellij.openapi.ui.TestInputDialog
+
+/**
```
Comment (hunterwerlla) [inline jetbrains-core/tst-201-202/software/aws/toolkits/jetbrains/ui/TestDialogService.kt:10]: `FIX_WHEN_MIN_IS_203 remove` ?

File: jetbrains-rider/protocol/model/daemon/LambdaDaemonModel.kt [hunk 3]
```diff
@@ -46,6 +46,9 @@ object LambdaDaemonModel : Ext(SolutionModel.Solution) {
     }
 
     init {
+        setting(Kotlin11Generator.Namespace, "software.aws.toolkits.jetbrains.protocol")
+        setting(CSharp50Generator.Namespace, "AWS.Toolkit.Rider.Model")
```
Comment (hunterwerlla) [inline jetbrains-rider/protocol/model/daemon/LambdaDaemonModel.kt:50]: Do we need comments on what these mean?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/DownloadLogStream.kt (hunk 1 above)
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/DownloadLogStream.kt:144]: Nah, they overloaded the method leading to us having to specify the type. I dont think we will ever remove it.

File: jetbrains-rider/protocol/model/daemon/LambdaDaemonModel.kt (hunk 3 above)
Comment (abrooksv) [reply jetbrains-rider/protocol/model/daemon/LambdaDaemonModel.kt:50]: ya im still tweaking them. It broke 201,202

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/DownloadLogStream.kt (hunk 1 above)
Comment (hunterwerlla) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/DownloadLogStream.kt:144]: Makes sense

File: jetbrains-rider/protocol/model/daemon/LambdaDaemonModel.kt (hunk 3 above)
Comment (abrooksv) [reply jetbrains-rider/protocol/model/daemon/LambdaDaemonModel.kt:50]: Okay fixed, they are the kotlin package and C# namespace to put our code into

**Review by hunterwerlla — APPROVED** [review]

**Comment by abrooksv** [comment]
Latest 203 Rider snapshot fixes the class loader issues.

**Pull request merged**